### PR TITLE
fix issue #901:add xz to rh6 netboot pkglist; do not mount /proc,/sys and /dev during genimage for rh6; do not add syslog module for dracut to generate initrd on rh6

### PR DESCRIPTION
**Pull request by immarvin**
fix issue #https://github.com/xcat2/xcat-core/issues/901
add xz to rh6 netboot pkglist;
do not mount /proc,/sys and /dev during genimage for rh6;
do not add syslog module for dracut to generate initrd on rh6

### CODE REVIEW

File: xCAT-server/share/xcat/netboot/rh/genimage [hunk 1]
```diff
@@ -73,28 +73,47 @@ sub xdie {
...
 }
 
+
+#helper subroutine to get the major release number 
+#of a osver
+sub majversion {
+    my $version=shift;
+    my $majorrel;
+
+    if($osver =~ /$\D*(\d*)[.\d]*$/){
```
Comment (daniceexi) [inline xCAT-server/share/xcat/netboot/rh/genimage]: The first **$** should be a typo 
~~~~
if($osver =~ /$\D*(\d*)[.\d]*$/){
~~~~

File: xCAT-server/share/xcat/netboot/rh/genimage [hunk 2]
```diff
@@ -73,28 +73,47 @@ sub xdie {
...
-    #postinstall script of package installation
-    #might access the /proc, /sys and /dev filesystem
-    #mount them from host read-only
-    system("mkdir -p $rootimage_dir/proc");
-    system("mount proc $rootimage_dir/proc -t proc -o ro");
-    system("mkdir -p $rootimage_dir/sys");
-    system("mount sysfs $rootimage_dir/sys -t sysfs -o ro");
-    system("mkdir -p $rootimage_dir/dev");
-    system("mount devtmpfs $rootimage_dir/dev -t devtmpfs -o ro");
+    if(majversion($osver) > 6){
```
Comment (daniceexi) [inline xCAT-server/share/xcat/netboot/rh/genimage]: Please add more comment for why rh6 does not work

File: xCAT-server/share/xcat/netboot/rh/genimage [hunk 3]
```diff
@@ -73,28 +73,47 @@ sub xdie {
    die @_;
 }
 
+
+#helper subroutine to get the major release number 
+#of a osver
+sub majversion {
+    my $version=shift;
```
Comment (daniceexi) [inline xCAT-server/share/xcat/netboot/rh/genimage]: pay attention to the format: should have space between **=**

**Comment by immarvin** [comment]
updated

**Comment by daniceexi** [comment]
I have reviewed the change and agree to merge.

**Pull request merged**